modules/nextflow/src/main/groovy/nextflow/cli/CmdRun.groovy

```diff
@@ -481,7 +481,7 @@ class CmdRun extends CmdBase implements HubOptions {
             if( offline )
                 throw new AbortOperationException("Unknown project `$repo` -- NOTE: automatic download from remote repositories is disabled")
             log.info "Pulling $repo ..."
-            def result = manager.download()
+            def result = manager.download(revision)
```

Comment:
Not sure about this, because there's an explicit checkout immediately after. But I guess the revision is needed here exactly to prevent this issue

Reply:
exactly
download function tries to validate if it's a valid repo but currently is doing checking the master branch